README.md

```diff
@@ -1,135 +1,170 @@
+To load pre-trained diarization models from the Hub, you'll first need to accept the terms-of-use for the following two models:
+1. [pyannote/segmentation-3.0](https://hf.co/pyannote/segmentation-3.0)
+2. [pyannote/speaker-diarization-3.1](https://hf.co/pyannote/speaker-diarization-3.1)
+
+And subsequently use a Hugging Face authentication token to log in with: 
 
 ```
-pip install -U "huggingface_hub[cli]"
-hugging-cli login
+huggingface-cli login
 ```
 
 ## Train
 
-When fine-tuning a `pyannote` segmentation model on a given dataset, be sure to specify this: 
-
-- `dataset_name`: Specify a dataset from the Hub on which to fine-tune your model.  
-- `dataset_config_name`:  If the dataset contains multiple language subsets, select the language ID of the subset you want to train on.
-- `train_split_name`: Specify which dataset split is to be used for training. Default is 'train'. 
-- `eval_split_name`: Specify which dataset split is to be used for validation. Default is 'validation'. 
-
-If the data set doesn't already contain a train and a validation split, you can automatically split it into train-val-test (90-10-10) using: 
-
-- `do_split_on_subset`: Specify the subset of the dataset you want to split into train-val-set.
+The script [`train_segmentation.py`](train_segmentation.py) can be used to pre-process a diarization dataset and subsequently
+fine-tune the `pyannote` segmentation model. In the following example, we fine-tune the segmentation model on the Japanese
+subset of the [CallHome dataset](https://huggingface.co/datasets/diarizers-community/callhome), a conversational dataset
+between native speakers:
 
-```
-python3 train_segmentation.py
+```bash
+python3 train_segmentation.py \
     --dataset_name=diarizers-community/callhome \
-    --dataset_config_name=jpn \ 
-    --do_split_on_subset=data \
+    --dataset_config_name=jpn \
+    --split_on_subset=data \
     --model_name_or_path=pyannote/segmentation-3.0 \
-    --output_dir=diarizers-community/speaker-segmentation-fine-tuned-callhome-jpn \
-    --preprocessing_num_workers=2 \ 
+    --output_dir=./speaker-segmentation-fine-tuned-callhome-jpn \
     --do_train \
-    --do_eval \ 
-    --learning_rate=1e-3 \ 
-    --num_train_epochs=1 \
-    --per_device_train_batch_size=32 \ 
-    --per_device_eval_batch_size=32 \ 
+    --do_eval \
+    --learning_rate=1e-3 \
+    --num_train_epochs=1 \
+    --per_device_train_batch_size=32 \
+    --per_device_eval_batch_size=32 \
     --evaluation_strategy=epoch \
-    --save_strategy=epoch \  
-    --dataloader_num_workers=2 \ 
-    --logging_steps=200 \
+    --save_strategy=epoch \
+    --preprocessing_num_workers=2 \
+    --dataloader_num_workers=2 \
+    --logging_steps=5 \
     --load_best_model_at_end \
     --push_to_hub
 ```
 
-## Inference
+On a single NVIDIA RTX 24GB GPU, training takes approximately 2 minutes and improves the diarization error rate (DER) 
```

Comment:
Have you inspected the training logs for these fine-tuning runs? I modified the above script to log the training loss at every step and report to wandb: https://wandb.ai/sanchit-gandhi/huggingface/runs/vfvts5lf/workspace?nw=nwusersanchitgandhi

It shows that the training loss doesn't really decrease over the course of training. This is because:
1. We only train for 1 epoch (=300 training steps)
2. We linearly decrease the learning rate over these training steps -> the learning rate is only high for 200 steps, after which it has been decayed to a low value

We're simply not training the model enough (e.g. long enough and with enough time when the learning rate is high). We could get much better performance by inspecting these logs and updating the HPs accordingly